### PR TITLE
Create SpearmanCorrelation.py

### PR DESCRIPTION
**Pull request by kibamin**

### CODE REVIEW

File: scripts/Word Simirarity Assessment/SpearmanCorrelation.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Compute Spearman’s rank correlation coefficient between this score and the human judgments
+# by Aamin Abbasi
+
+import pandas as pd
+import numpy as np
+from scipy.stats import spearmanr
+
+# get dataset name
+filename = 'SemanticSimilarityDataset.csv.xlsx'
```
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: I don't want to support Excel file we need only CSV file it is very simple.

File: scripts/Word Simirarity Assessment/SpearmanCorrelation.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+f_extns = filename.split(".")
+
+# if similarity dataset format is '.xlsx' with four columns (word1, word2, similarity, human_judgment)
+if(f_extns[-1] == 'xlsx'):
```
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: Only CSV please

File: scripts/Word Simirarity Assessment/SpearmanCorrelation.py (hunk 1 above)
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: You have to scan the data/similarity folder and load any CSV file there.  Don't hard code the file name 
Also for each file you have to create a same filename output CSV

File: scripts/Word Simirarity Assessment/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+This code compute Spearman’s rank correlation coefficient between this score and the human judgments.
```
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/README.md]: Please document which packages need to be installed for this code to work

File: scripts/Word Simirarity Assessment/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+INPUT : Set similarity dataset name with formats : ( ".xlsx" OR ".txt" OR ".csv" ) and 4 columns that is : word1 | word2 | sim | human_judgment
```
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/README.md]: Only CSV!

Comment (sehsanm) [inline scripts/Word Simirarity Assessment/README.md]: What is the difference between sim and human judgement ?

File: scripts/Word Simirarity Assessment/SpearmanCorrelation.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+# if similarity dataset format is '.CSV' or '.txt'
+else:
+    similarity_dataset = np.genfromtxt(filename, delimiter=',')
+    similarity_rate = similarity_dataset[:,2].tolist()
+    human_judgment = similarity_dataset[:,3].tolist()
+
+
+corr, p_value = spearmanr(similarity_rate, human_judgment)
+print("Spearman's rank correlation coefficient : ",corr)
```
Comment (sehsanm) [inline scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: Where are you getting the output from the model ?  I'm a bit confused

**Comment by sehsanm** [comment]
Please also change the folder name from "Word Similarity Assessment" to "similarity" Folders with space in their names are not good.

**Comment by sehsanm** [comment]
Also my view is that you have to first write a function that accepts a model as input and then returns the score. The code that you have written is not written with a library mentality it is a once off running code.

File: scripts/Word Simirarity Assessment/SpearmanCorrelation.py (hunk 1 above)
Comment (kibamin) [reply scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: > Any update or progress ?
> Please let me know when you can finish this

من کامنت گذاشتم هفته پیش ولی هیچ جوابی نگرفتم
this comment : https://github.com/sehsanm/embedding 
benchmark/pull/31#discussion_r243044489
 اگه مدل باید دریافت کنم ممنون میشم اگه راهنمایی کنید مدل دقیقا به چه صورته؟
ولی به نظرم کار بهتر اینکه به جای دریافت مدل به عنوان ورودی بهتره که قبلش فایلها رو به اون مدل بدیم و 
بزنم؟ correlation  خروجی شون من 
نظر شما چیه؟

Comment (kibamin) [reply scripts/Word Simirarity Assessment/SpearmanCorrelation.py]: اگه مدل دریافت کنم پس نیازمند همچین ماژولی هستم که شباهت 2 تا کلمه رو حساب کنه
model.smilarity("word1","word2")

درست میگم؟

**Pull request closed without merging**